docs/source/dev-docs.rst

```diff
@@ -127,6 +127,95 @@ to run each do the following:
+which requires docker. OpenDCS-XML only depends on the file system.
+
+to run either use the following command::
+
+    ant integration-test -Dno.doc=true -Dopendcs.test.engine=OpenDCS-XML
+    # or 
+    ant integration-test -Dno.doc=true -Dopendcs.test.engine=OpenDCS-Postgres
+
+Adding tests
+------------
+
+New classes, or methods to existing classes, should go under :code:`org.opendcs.regression_tests`
+
+:code:`AppTestBase` contains for members accessible to your tests
+
+
++--------------------------------------------+--------------------------------+
+|Member Variable                             |Description                     |
++============================================+================================+
+|@SystemStub\                                |variables from \                |
+|protected final EnvironmentVariables \      |System.getenv \                 |
+|environment = new EnvironmentVariables();   |that applications will see.     |
++--------------------------------------------+--------------------------------+
+|@SystemStub\                                |variables from \                |
+|protected final SystemProperties \          |System.getProperty \            |
+|properties = new SystemProperties();        |that applications will see.     |
++--------------------------------------------+--------------------------------+
+|@SystemStub\                                |Used to trap System.exit        |
+|protected final SystemExit \                |calls to allow testing          |
+|exit = new SystemExit();                    |without aborting the test run   |
++--------------------------------------------+--------------------------------+
+|@ConfiguredField                            |Instance of the                 |
+|protected Configuration configuration;      |:code:`Configuration` that was  |
+|                                            |create for this run. Contains   |
+|                                            |reference to user.properties and|
+|                                            | other specific information     |
++--------------------------------------------+--------------------------------+
+ 
+Extension and other Junit information
+-------------------------------------
+
+The :code:`OpenDCSTestConfigExtension`, if it knows about a given type, will inject 
+an instance of any field annotated with :code:`@ConfiguredField` as seen in the table 
+above for the configuration.
+
+The only other injected field is a :code:`TimeSeriesDb` which is Provided by the Configuration
+and will already be valid and can be used directly for things like testing DaoObjects or null which 
+indicates the implementation under test doesn't use the any of the timeseries database components.
+
+The sample :code:`LoadingAppDaoTestIT` uses the :code:`@EnableIfSql` :code:`ExecutionCondition`
+ to determine if the test should be run or not.
+
```

Comment:
The sample :code:`LoadingAppDaoTestIT` uses the :code:`@EnableIfSql`  **_annotation that extends from Junit's_**  :code:`ExecutionCondition`
 to determine if the test should be run or not.